### PR TITLE
feat: Apply multiple files message item body

### PR DESCRIPTION
**Pull request by liamcho**
Feature: [UIKIT-4346](https://sendbird.atlassian.net/browse/UIKIT-4346)
Figma: https://www.figma.com/file/SVbXU00FhjztekD8AiVukK/UIKit_Work-file_React?node-id=1406%3A29078&mode=dev

Screen Shot 2023-08-29 at 5 11 24 PM

[UIKIT-4289]: https://sendbird.atlassian.net/browse/UIKIT-4289?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[UIKIT-4346]: https://sendbird.atlassian.net/browse/UIKIT-4346?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/ui/MultipleFilesMessageItemBody/index.tsx [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+  className?: string | Array<string>;
+  message: MultipleFilesMessage;
+  isByMe?: boolean;
+  mouseHover?: boolean;
+  isReactionEnabled?: boolean;
+  truncateLimit?: number;
+}
+
+// Full width is 400px. There are two items in a row which means image width is 200px. padding is 4px and gap is 4px.
+const MULTIPLE_FILES_IMAGE_SIDE_LENGTH = 'calc(200px - 6px)';
```
Comment (AhyoungRyu) [inline src/ui/MultipleFilesMessageItemBody/index.tsx:19]: Q. how 4px came out from "padding is 4px and gap is 4px"?

File: src/ui/MultipleFilesMessageItemBody/index.tsx [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+        isReactionEnabled={isReactionEnabled}
+      >
+        {
+          fileInfoList.map((fileInfo: UploadedFileInfo, index: number) => {
+            return <div
+              onClick={() => setCurrentIndex(index)}
+              style={{
+                cursor: 'pointer',
+              }}
+              key={`image-renderer-${index}`}
```
Comment (AhyoungRyu) [inline src/ui/MultipleFilesMessageItemBody/index.tsx:73]: It's better to use a really unique id if UploadedFileInfo has id like field. 
https://react.dev/learn/rendering-lists#why-does-react-need-keys <- this(especially Pitfall section) might be helpful why always better not to use index as a list key.

File: src/ui/MultipleFilesMessageItemBody/index.tsx (hunk 1 above)
Comment (liamcho) [reply src/ui/MultipleFilesMessageItemBody/index.tsx:19]: ah i knew ill get this question lol. its like below
[left padding] [left image] [gap] [right image] [right padding]

File: src/ui/FileViewer/index.scss [hunk 3]
```diff
@@ -104,11 +106,7 @@ $file-viewer-img-max-width: calc(100% - #{$file-viewer-slide-buttons-side-length
     }
     .sendbird-fileviewer__content__img__multi {
       max-width: $file-viewer-img-max-width;
-      max-height: 100%;
-      @include mobile() {
-        max-width: 100%;
-        max-height: 100%;
-      }
```
[the remark this answers is not in the document]
Comment (HoonBaek) [reply src/ui/FileViewer/index.scss:111]: It's agreed upon with Gabie, right? @liamcho

[the remark this answers is not in the document]
Comment (liamcho) [reply src/ui/FileViewer/index.scss:111]: yes that mobile should have been removed after we changed design from swipe to slide button

File: src/ui/ImageGrid/index.tsx [hunk 4]
```diff
@@ -1,16 +1,28 @@
...
 export default function ImageGrid({
   children,
+  className,
+  message,
+  isReactionEnabled,
 }: ImageGridProps): ReactElement {
   return (
     <div className='sendbird-image-grid-wrap'>
-      <div className='sendbird-image-grid'>
+      <div className={getClassName([
```
Comment (HoonBaek) [inline src/ui/ImageGrid/index.tsx:21]: LOL, actually you don't need to accept the className as `Array<string>` type
It was my bad practice a long long time ago
Anyway this is fine too

File: src/ui/MultipleFilesMessageItemBody/index.tsx (hunk 1 above)
Comment (HoonBaek) [reply src/ui/MultipleFilesMessageItemBody/index.tsx:19]: I also can't understand well, will it be
[4px] [200px] [4px] [200px] [4px]?

Then the maximum width should be 412px

File: src/ui/MultipleFilesMessageItemBody/index.tsx [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+  message: MultipleFilesMessage;
+  isByMe?: boolean;
+  mouseHover?: boolean;
+  isReactionEnabled?: boolean;
+  truncateLimit?: number;
+}
+
+// Full width is 400px. There are two items in a row which means image width is 200px. padding is 4px and gap is 4px.
+const MULTIPLE_FILES_IMAGE_SIDE_LENGTH = 'calc(200px - 6px)';
+const MULTIPLE_FILES_IMAGE_BORDER_RADIUS = '6px';
```
Comment (HoonBaek) [inline src/ui/MultipleFilesMessageItemBody/index.tsx:20]: Can we keep these constants in the const.ts file?

File: src/ui/MultipleFilesMessageItemBody/index.tsx (hunk 1 above)
Comment (liamcho) [reply src/ui/MultipleFilesMessageItemBody/index.tsx:19]: i know this is confusing but this was the simplest way for me.
so the image grid width should be 400px fixated.
so each image's side length should have 200px including its side padding and gap.
so 200 - (4 / 2) - (4 / 2) - (4 / 2) = 200 - 6

File: src/ui/MultipleFilesMessageItemBody/index.tsx [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+  return (
+    <>
+      {
+        currentIndex > -1 && <FileViewer
+          message={message}
+          currentIndex={currentIndex}
+          onClickLeft={onClickLeft}
+          onClickRight={onClickRight}
+          onClose={onClose}
+        />
```
[the remark this answers is not in the document]
Comment (liamcho) [reply src/ui/MultipleFilesMessageItemBody/index.tsx]: yeah no problem

File: src/ui/MultipleFilesMessageItemBody/index.tsx [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+              }}
+              key={`multiple-files-image-renderer-${index}-${fileInfo.plainUrl}`}
+            >
+              <ImageRenderer
+                url={fileInfo.url}
+                width={MULTIPLE_FILES_IMAGE_SIDE_LENGTH}
+                height={MULTIPLE_FILES_IMAGE_SIDE_LENGTH}
+                borderRadius={MULTIPLE_FILES_IMAGE_BORDER_RADIUS}
+                defaultComponent={
+                  <Icon type={IconTypes.ADD} fillColor={IconColors.PRIMARY}/>
```
[the remark this answers is not in the document]
Comment (liamcho) [reply src/ui/MultipleFilesMessageItemBody/index.tsx:83]: how do i do that?

[the remark this answers is not in the document]
Comment (HoonBaek) [reply src/ui/MultipleFilesMessageItemBody/index.tsx:78]: Here, you can see the default values are 26px
https://github.com/sendbird/sendbird-uikit-react/blob/main/src/ui/Icon/index.jsx#L196

File: src/ui/MultipleFilesMessageItemBody/index.tsx (hunk 1 above)
Comment (AhyoungRyu) [reply src/ui/MultipleFilesMessageItemBody/index.tsx]: Or we can create a const for 4 and 2 too for the better readability. Not sure the calculation is correct as your intention but, the idea is like below. 

```
const MESSAGE_CONTAINER_WIDTH = 400;
const PADDING = 4;

const MULTIPLE_FILES_IMAGE_SIDE_LENGTH = `calc(${MESSAGE_CONTAINER_WIDTH / 2} - 3*${PADDING / 2})`
```

File: src/ui/MultipleFilesMessageItemBody/const.ts [hunk 8]
```diff
@@ -0,0 +1,8 @@
+const IMAGE_GRID_WIDTH = 400;
+const PADDING = 4;
+const GAP = 4;
+
+export const MULTIPLE_FILES_IMAGE_SIDE_LENGTH = `calc(${IMAGE_GRID_WIDTH / 2}px - 
+${2 * (PADDING / 2)}px - ${GAP / 2}px)`;
+export const MULTIPLE_FILES_IMAGE_BORDER_RADIUS = '6px';
+export const MULTIPLE_FILES_IMAGE_THUMBNAIL_SIDE_LENGTH = '40px';
```
Comment (HoonBaek) [inline src/ui/MultipleFilesMessageItemBody/const.ts:8]: 👍

File: src/ui/MultipleFilesMessageItemBody/index.scss [hunk 9]
```diff
@@ -0,0 +1,21 @@
...
+  justify-content: center;
+  border-radius: 12px;
+  align-items: center;
+  @include themed() {
+    background-color: t(bg-2);
+  }
+}
+
+.sendbird-multiple-files-image-renderer__thumbnail__placeholder__icon {
+}
```
Comment (HoonBaek) [inline src/ui/MultipleFilesMessageItemBody/index.scss:21]: we can remove this, right?

**Review by HoonBaek — APPROVED** [review]
Thank you!

File: src/modules/App/stories/index.stories.js [hunk 10]
```diff
@@ -265,7 +266,7 @@ export const Korean = () => fitPageSize(
 export const user1 = () => fitPageSize(
   <App
     appId={appId}
-    userId={array[0]}
+    userId={'sendbird1'}
```
Comment (HoonBaek) [inline src/modules/App/stories/index.stories.js:269]: just for reminding~ 😉

File: src/ui/MultipleFilesMessageItemBody/index.scss (hunk 9 above)
Comment (liamcho) [reply src/ui/MultipleFilesMessageItemBody/index.scss]: yep removed from child PR

File: src/ui/MessageContent/index.tsx [hunk 11]
```diff
@@ -239,7 +240,10 @@ export default function MessageContent({
       </div>
       {/* middle */}
       <div
-        className="sendbird-message-content__middle"
+        className={isMobile
+          ? ''
+          : 'sendbird-message-content__middle'
+        }
```
[the remark this answers is not in the document]
Comment (liamcho) [reply src/ui/MessageContent/index.tsx:246]: @HoonBaek @AhyoungRyu

[the remark this answers is not in the document]
Comment (HoonBaek) [reply src/ui/MessageContent/index.tsx:246]: Let me take a look at this :D

**Pull request merged**